staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -151,11 +212,100 @@ func WithTypeAndObjectMeta(s *schema.Structural) *schema.Structural {
 	props["metadata"] = schema.Structural{
 		Generic: schema.Generic{Type: "object"},
 		Properties: map[string]schema.Structural{
-			"name": stringType,
+			"name":         stringType,
 			"generateName": stringType,
 		},
 	}
 	result.Properties = props
 
 	return result
-}
\ No newline at end of file
+}
+
+// estimateMinSizeJSON estimates the minimum size in bytes of the given schema when serialized in JSON.
+// minLength/minProperties/minItems is not currently taken into account, so the estimate is a little
+// smaller than the actual smallest size.
+func estimateMinSizeJSON(s *schema.Structural) int64 {
+	switch s.Type {
+	// take a comma into account for all cases since we'll generally call this function for arrays/maps
+	case "boolean":
+		// true,
+		return 5
+	case "number", "integer":
+		// 0,
+		return 2
+	case "string":
+		if s.ValueValidation != nil && s.ValueValidation.Format != "" && s.ValueValidation.Format != "byte" {
+			switch s.ValueValidation.Format {
+			case "duration":
+				return minDurationSizeJSON
+			case "date":
+				// add 1 to make room for a comma; since we're in estimateMinSizeJSON we need to account for
+				// this date being an array element
+				return dateSizeJSON + 1
+			case "date-time":
+				return minDatetimeSizeJSON
+			}
+		}
+		// "",
+		return 3
+	case "array":
+		// [],
+		return 3
+	case "object":
+		objSize := int64(3) // {},
+		// sum of all non-optional properties
+		if s.ValueValidation != nil {
+			for _, propName := range s.ValueValidation.Required {
+				prop := s.Properties[propName]
+				// add 3, 2 for quotations around the property name and 1 for the colon
+				objSize += int64(len(propName)) + estimateMinSizeJSON(&prop) + 3
+			}
+		}
+		return objSize
+	}
+	// TODO(DangerOnTheRanger): better error handling (We should never get here in normal operation)
```

Comment:
I think it's okay to leave this handled the way it is. Adding a new type to OpenAPIv3 is not allowed, it would break many things. I'd just leave this comment along the lines of "This should never happen since all OpenAPIv3 types are handled"

Reply:
Makes sense, I'll adjust the comment and mark this as resolved once the change is pushed.